hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatReader.java

```diff
@@ -172,21 +171,35 @@ public boolean hasNext() {
     }
   }
 
-  private boolean readMagic() throws IOException {
+  // TODO : separate reading of markers once idempotency is resolved.
```

Comment:
@prazanna I think a clean separation of read..() methods is possible if we can make read idempotent (which seems difficult without seeking the inputstream back to a previous pos which increase I/O for every read). Thoughts ?

Reply:
Should we model End Marker as a special COMMAND_BLOCK ? That way you do not have 2 markers you look for at each block and I think that way it is also clean instruction to the scanner to stop here.

Reply:
Yes, I think using a special COMMAND_BLOCK makes sense here. I'll update the PR.

Reply:
@prazanna Although COMMAND_BLOCK is a nice end marker, it's nasty to use it in the LogWriter. Since we want to prevent data being written to a sealed log file, the writer will now need to read LogBlocks. Thoughts ?

Reply:
@n3nash why would that be. if we know the last block has to be COMMAND_BLOCK with end marker for a file to be sealed, could n't we reverse derive the seek pos from the file length?